services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/KafkaMessageCollector.java

```diff
@@ -109,35 +110,46 @@ public void startUp(FloodlightModuleContext moduleContext) throws FloodlightModu
         commandProcessor.init(moduleContext);
         pingService.init(moduleContext);
 
-        initConsumer(moduleContext);
-    }
-
-    private void initConsumer(FloodlightModuleContext moduleContext) {
         KafkaConsumerConfig consumerConfig = configService.getConsumerConfig();
+        KafkaTopicsConfig topicsConfig = configService.getTopics();
+        ConsumerContext context = new ConsumerContext(moduleContext, topicsConfig);
+
+        ExecutorService lowPriorityExecutor = buildExecutorWithNoQueue(consumerConfig.getExecutorCount());
+        initConsumer(consumerConfig, context, topicsConfig.getSpeakerTopic(), lowPriorityExecutor);
+        initConsumer(consumerConfig, context, topicsConfig.getSpeakerFlowTopic(), lowPriorityExecutor);
+        initConsumer(consumerConfig, context, topicsConfig.getSpeakerFlowPingTopic(), lowPriorityExecutor);
 
-        logger.info("config - executor threads = {}", consumerConfig.getExecutorCount());
+        ExecutorService discoCommandExecutor = buildExecutorWithNoQueue(consumerConfig.getExecutorCount());
```

Comment:
We should have separate config option to define thread count for each thread pool.